graphql_compiler/macros/macro_expansion.py

```diff
@@ -292,9 +296,15 @@ class name -> (macro edge name -> MacroEdgeDescriptor)
                     _check_that_expansion_directives_are_supported(selection_ast)
                     macro_edge_descriptor = macro_edges_at_this_type[field_name]
 
+                    # Sanitize the macro, making sure it doesn't use any taken tag names.
+                    macro_tag_names = get_all_tag_names(macro_edge_descriptor.expansion_ast)
+                    name_change_map = generate_disambiguations(tag_names, macro_tag_names)
+                    tag_names |= set(name_change_map.values())
```

Comment:
Nit: how about `tag_names.update(name_change_map.values())`? It saves a set construction by just walking the iterable directly.